src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1477,7 +1477,9 @@ private string GetLocalDebugName(ILocalSymbolInternal local, out LocalDebugId lo
             // only user-defined locals should be named during lowering:
             Debug.Assert((local.Name == null) == (localKind != SynthesizedLocalKind.UserDefined));
 
-            if (!localKind.IsLongLived())
+            // Skip naming locals generated by instrumentation as they might not have a local syntax offset.
+            // Locals generated by instrumentation might exist in methods which do not contain a body (auto property initializers).
+            if (!localKind.IsLongLived() || localKind == SynthesizedLocalKind.InstrumentationPayload)
             {
```

Comment:
It might be that I just don't understand the scenario, but how does this break the existing code? 

It should just extend it so that locals created by Instrumentation don't have a local debug name, correct?